### PR TITLE
Fix compile error with ZONESTAR_LCD+NO_LCD_MENUS

### PR DESCRIPTION
**Pull request by Ludy87**
compile error `#define NO_LCD_MENUS` and `#define ZONESTAR_LCD`

```c++
In function `MarlinUI::update_buttons()':
<artificial>:(.text+0x138): undefined reference to `MarlinUI::encoderDirection'
<artificial>:(.text+0x144): undefined reference to `MarlinUI::encoderDirection'
In function `MarlinUI::update()':
<artificial>:(.text+0x5d2): undefined reference to `MarlinUI::encoderDirection
```

### CODE REVIEW

File: Marlin/src/lcd/ultralcd.h [hunk 1]
```diff
@@ -27,7 +27,7 @@
   #include "../libs/buzzer.h"
 #endif
 
-#define HAS_ENCODER_ACTION (HAS_LCD_MENU || ENABLED(ULTIPANEL_FEEDMULTIPLY))
+#define HAS_ENCODER_ACTION (HAS_LCD_MENU)
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.h]: The purpose of this conditional is to indicate that there's something in the active LCD that can make use of the encoder knob. That includes: menu navigation and value editing, and changing the feed multiply value on the status screen, which is available even when `NO_LCD_MENUS` is enabled.

File: Marlin/src/lcd/ultralcd.cpp [hunk 2]
```diff
@@ -814,7 +809,7 @@ void MarlinUI::update() {
           break;
       } // switch
 
-      #if ENABLED(ADC_KEYPAD)
+      #if ENABLED(ADC_KEYPAD) && HAS_ENCODER_ACTION
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.cpp]: The ADC keypad is either enabled or it's not. And note that these keypads allow for functions that don't require LCD menus or feed multiply to be available. For example, some keypads have a dedicated "home" button that runs `G28`. So this combination isn't really sensible.

File: Marlin/src/lcd/ultralcd.cpp [hunk 3]
```diff
@@ -708,17 +708,12 @@ void MarlinUI::update() {
...
       #if ENABLED(REPRAPWORLD_KEYPAD)
 
-        if (
-          #if ENABLED(ADC_KEYPAD)
-            handle_keypad()
-          #else
-            handle_keypad()
-          #endif
-        ) {
+        if (handle_keypad()) {
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.cpp:711]: Ha! I wonder how I let this one slip by. Zzzzz…

File: Marlin/src/lcd/ultralcd.h (hunk 1 above)
Comment (Ludy87) [reply Marlin/src/lcd/ultralcd.h]: Okay, but if the code remains snippets, then compiled does not work.

```c++
In function `MarlinUI::update_buttons()':
<artificial>:(.text+0x138): undefined reference to `MarlinUI::encoderDirection'
<artificial>:(.text+0x144): undefined reference to `MarlinUI::encoderDirection'
In function `MarlinUI::update()':
<artificial>:(.text+0x5d2): undefined reference to `MarlinUI::encoderDirection'
```

File: Marlin/src/lcd/ultralcd.cpp (hunk 2 above)
Comment (Ludy87) [reply Marlin/src/lcd/ultralcd.cpp]: ok i understand

File: Marlin/src/lcd/ultralcd.cpp (hunk 3 above)
Comment (Ludy87) [reply Marlin/src/lcd/ultralcd.cpp:711]: That happens in the late hour

**Comment by thinkyhead** [comment]
As a more comprehensive approach, I've added a condition to differentiate between controllers that simply have digital buttons and encoders that include an encoder knob with a rolling two-bit state. The Zonestar has no encoder, but the RepRapWorld Keypad has an encoder knob. This way most of the existing conditions remain valid, but the code to read the encoder wheel can be skipped for `ADC_KEYPAD`.

**Pull request merged**